paradox-nakadi-consumer-sqs-error-handler/src/main/java/de.zalando.paradox.nakadi.consumer.sqserrorhandler/SQSFailedEventSource.java

```diff
@@ -0,0 +1,115 @@
+        this.amazonSQS = amazonSQS;
+        this.objectMapper = objectMapper;
+        this.queueUrl = getQueueUrl(sqsConfig.getQueueName());
+    }
+
+    @Override
+    public String getEventSourceName() {
+        return EVENT_SOURCE_NAME;
+    }
+
+    @Override
+    public Optional<SQSFailedEvent> getFailedEvent() {
+        return fetchEvent().map(this::mapToSQSFailedEvent);
+    }
+
+    @Override
+    public void commit(final SQSFailedEvent sqsFailedEvent) {
+
+        final DeleteMessageResult deleteMessageResult = amazonSQS.deleteMessage(queueUrl,
+                sqsFailedEvent.getReceiptHandle());
+
+        if (deleteMessageResult.getSdkHttpMetadata().getHttpStatusCode() != 200) {
+            throw new IllegalStateException("The event could not removed from the queue.");
+        }
+    }
+
+    @Override
+    public long getSize() {
+
+        final GetQueueAttributesRequest getQueueAttributesRequest = new GetQueueAttributesRequest(queueUrl,
+                Collections.singletonList(QueueAttributeName.ApproximateNumberOfMessages.name()));
+
+        final GetQueueAttributesResult queueAttributes = amazonSQS.getQueueAttributes(getQueueAttributesRequest);
+
+        if (queueAttributes.getSdkHttpMetadata().getHttpStatusCode() != 200) {
+            throw new IllegalStateException("ApproximatelyTotalNumberOfFailedEvents could not retrieved from SQS.");
+        }
+
+        if (queueAttributes.getAttributes() != null) {
+            return Long.valueOf(queueAttributes.getAttributes().getOrDefault(
+                        QueueAttributeName.ApproximateNumberOfMessages.name(), "0"));
+        } else {
+            return 0L;
+        }
+    }
+
+    private Optional<Message> fetchEvent() {
+
+        final ReceiveMessageResult receiveMessageResult = amazonSQS.receiveMessage(queueUrl);
+
+        if (receiveMessageResult.getMessages() != null && !receiveMessageResult.getMessages().isEmpty()) {
+            return Optional.of(receiveMessageResult.getMessages().get(0));
+        } else {
+            return Optional.empty();
+        }
+    }
+
+    private SQSFailedEvent mapToSQSFailedEvent(final Message message) {
+        try {
+            final FailedEvent failedEvent = objectMapper.readValue(message.getBody(), FailedEvent.class);
+            final SQSFailedEvent sqsFailedEvent = new SQSFailedEvent(failedEvent);
+            sqsFailedEvent.setId(message.getMessageId());
+            sqsFailedEvent.setReceiptHandle(message.getReceiptHandle());
+            return sqsFailedEvent;
+        } catch (final Exception e) {
+            throw new IllegalStateException(String.format(
+                    "Exception occurred during deserialization. Message id = [%s]", message.getMessageId()));
+        }
+    }
+
+    private String getQueueUrl(final String queueName) {
+
+        final GetQueueUrlResult queueUrl = amazonSQS.getQueueUrl(queueName);
+
+        if (queueUrl.getSdkHttpMetadata().getHttpStatusCode() != 200) {
```

Comment:
AmazonSQSClient does error handling already (see `AmazonSQSClient.init()`). 

```
exceptionUnmarshallers.add(new QueueDoesNotExistExceptionUnmarshaller());
..
exceptionUnmarshallers.add(new StandardErrorUnmarshaller());
```

Checking the status code might not be needed. You might want to check if `queueUrl.getQueueUrl()` is non-null, though.